src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -657,6 +658,17 @@ public function handleBlockActorData(BlockActorDataPacket $packet) : bool{
 			}
 
 			$this->session->getLogger()->debug("Invalid sign update data: " . base64_encode($packet->nbt->getEncodedNbt()));
+		}elseif($block instanceof Beacon){
+			try{
+				$block->setPrimaryEffect($nbt->getInt("primary", 0));
+				$block->setSecondaryEffect($nbt->getInt("secondary", 0));
```

Comment:
The following things should be checked before activating the beacon:

- Whether or not the height of the pyramid allows the selected effects
- Whether the beacon inventory window contains a valid type of payment (ingots, diamonds, emeralds)

There should also be an event to allow controlling this from a plugin.